### ISSUE TITLE
Feature request: AngularJS 1.x revival

### ISSUE THREAD

**Issue by jamesharvey2** [ionic-team/ionic#14166]
**Ionic version:**  (check one with "x")
[X ] **1.x** (For Ionic 1.x issues, please use https://github.com/ionic-team/ionic-v1)
[ ] **2.x**
[ ] **3.x**

**I'm submitting a ...**  (check one with "x")
[ ] bug report
[X ] feature request
[ ] support request => Please do not submit support requests here, use one of these channels: https://forum.ionicframework.com/ or http://ionicworldwide.herokuapp.com/

Hello Ionic team,

I found out there are still a significant amount of companies using Angular JS 1.x. They don't want to upgrade to Angular 5 because AngularJS 1.x already works for them. 
To be honest, I see more companies looking for AngularJS 1.x developers. 
Thus, I see AngularJS 1.x as a valuable asset. I anticipate AngularJS will last more than four to five years. 
Is it possible for you guys to make Ionic 4 work with both Angular 5 and Angular JS 1.6? 
I started studying AngularJS 1.x and I think it would be great to be able to integrate the newest Ionic UI with 1.x. 
Maybe a silly idea but I found lots of established companies and individuals using 1.x.. 
Current demand in job market looks like 60% AngularJS, 40% Angular 2, 4, 5 so it makes sense to sustain angularJS compatibility.

**Comment by simonhaenisch** [comment]
I think it's theoretically possible with Ionic becoming framework-agnostic (see https://github.com/ionic-team/ionic/tree/master/angular for how they do it with angular). However I wouldn't expect any work on that and I'm actually quite happy that Ionic is becoming simpler again, maybe making it a lot easier to migrate old v1 apps. Or maybe you'll be able to write the "AngularJS specific building blocks on top of `@ionic/core`" yourself ;)

**Comment by jamesharvey2** [comment]
Angular 5 is technically better... but when so many companies are into 1.x, you may want to have 1.x Ionic 4 to find good opportunities and wealthy clients. 
I can study more React.js maybe..but that will be another long road when AngularJS offers similar amount of opportunities as react.js.

**Comment by jamesharvey2** [comment]
To guys thumbing down on this thread: some business people I'm talking about are millionaires...

**Comment by simonhaenisch** [comment]
I just checked the Ionic CLI changelog (https://github.com/ionic-team/ionic-cli/blob/d62c12b34293ef27fa03aa25470f547eaa05e99d/CHANGELOG.md#boom-breaking-changes) and it mentions that Ionic 1 functionality has been moved into a [`@ionic/v1-util`](https://github.com/ionic-team/ionic-cli/tree/master/packages/@ionic/v1-util) package. So it seems that CLI version 4 will still support Ionic 1.

**Comment by jamesharvey2** [comment]
Awesome. 
AngularJS is here to stay for four to five more years despite of its technical issues. Yes, it works for small businesses which don't need more than 2000 users on a single page.  and it's easy to use with just `<script>` tags.

**Comment by jamesharvey2** [comment]
My request was to make Ionic 4 UI available for AngularJS 1.x... so users will be able to use Ionic 4 with AngularJS 1.x but I think that'll take some time.

**Comment by simonhaenisch** [comment]
The Ionic 4 UI components will run as browser-native web components with a polyfill to support major browsers and IE11+ (you can read about it here: https://stenciljs.com/docs/browser-support). They are built using Stencil rather than a framework like Angular or React, thus you can use them with any other framework. However you would have to figure out yourself how to connect AngularJS and Stencil (which compiles from Typescript).

If I was part of Ionic I wouldn't support your request though cause the amount of work for it doesn't seem worth it at all. If people want to stick with v1 rather than going through a migration, then they can even with the latest CLI version, but then they have to live with outdated UI components and software practices. IMO now is one of the best times to consider migrating.

**Comment by jamesharvey2** [comment]
Yes, that's a great improvement.  Can't wait to try Ionic 4... and I'm sure I will use Ionic 4 for my own products and clients.

I understand but a lot of rich companies out there want to use traditional javascript, frameworks which already worked for them. If you help them, you can make your life better.

**Comment by mlynch** [comment]
First of all: whoever is thumbing this issue down, please don't. There's no place for that kind of negativity here.

Second of all, I agree that Angular 1 is still around. I actually am hopeful that Ionic 4 web components can target Angular 1 without us having to build and maintain a custom Angular 1 implementation. I built this proof of concept recently which shows how to use stencil web components (which Ionic 4 components are), in Angular 1: https://github.com/mlynch/stencil-transform-angularjs

So, basically: I think we'll be able to give Angular 1 devs some options, and we'd love to have their input and possibly contributions to test and take this further. We probably won't *officially* support it until we see how stable and appealing it is to developers.
